src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -1066,6 +1081,13 @@ class TreeRenderer extends Disposable implements ITreeRenderer<ITreeItem, FuzzyS
 			templateData.icon.style.backgroundImage = '';
 		}
 
+		if (!commandEnabled) {
+			templateData.icon.className = templateData.icon.className + ' disabled';
+			if (templateData.container.parentElement) {
+				templateData.container.parentElement.className = templateData.container.parentElement.className + ' disabled';
+			}
+		}
```

Comment:
Is all this needed given the changes in the `IconLabel`?

Reply:
The first statement makes sure to add change the opacity of the icon.
For the second part, as the HTML part of the IconLabel is only a portion of the entire TreeItem HTML, removing it would mean for example, that the left side of the label, just outside the Icon would show a pointer cursor when hovering.